### PR TITLE
Ziti Connection bridge

### PR DESCRIPTION
**Pull request by ekoby**
Ziti Connection bridge functions (see ziti.h) are utilities to simplify handling of application streams and IO file descriptors

### CODE REVIEW

File: includes/ziti/ziti.h [hunk 1]
```diff
@@ -800,7 +811,36 @@ extern int ziti_close_write(ziti_connection conn);
...
+ *
+ * @param conn
+ * @param input
+ * @param output
+ * @param close_cb
+ * @param ctx
+ * @return
+ */
+ZITI_FUNC
+extern int ziti_conn_bridge_fds(ziti_connection conn, uv_os_fd_t input, uv_os_fd_t output, void (*close_cb)(void *ctx), void *ctx);
```
Comment (smilindave26) [inline includes/ziti/ziti.h:843]: why a different close_cb than on the stream version (and why doesn't the stream bridge include a void *ctx)?

Comment (ekoby) [reply includes/ziti/ziti.h:843]: with the stream version we don't make any assumptions on how application want to close it -- so it is passed into the close callback. any custom data can be attached to that stream already `stream->data`

File: library/conn_bridge.c [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    br->data = br;
+    br->fdbr = fdbr;
+
+    uv_handle_set_data((uv_handle_t *) br->input, br);
+    ziti_conn_set_data(conn, br);
+
+    ziti_conn_set_data_cb(conn, on_ziti_data);
+    uv_read_start(br->input, bridge_alloc, on_input);
+
+    return ZITI_WTF;
```
Comment (smilindave26) [inline library/conn_bridge.c:114]: really return wtf?  expect to run forever (nobody calls uv_read_stop?)?  should return based on the return value from uv_read_start?

Comment (ekoby) [reply library/conn_bridge.c:114]: Umm... leftover TODO code. good catch

**Review by smilindave26 — APPROVED** [review]

**Pull request merged**